crates/typed-store/src/rocks/mod.rs

```diff
@@ -430,11 +437,46 @@ where
     }
 }
 
-/// Creates a default Rocksdb option.
+fn read_size_from_env(var_name: &str) -> Option<usize> {
+    match env::var(var_name) {
+        Ok(val) => match val.parse::<usize>() {
+            Ok(i) => Some(i),
+            Err(e) => {
+                info!(
+                    "Env var {} does not contain valid usize integer: {}",
+                    var_name, e
+                );
+                Option::None
+            }
+        },
+        Err(e) => {
+            info!("Env var {} is not set: {}", var_name, e);
+            Option::None
+        }
+    }
```

Comment:
1. `Option::None` is pedantic, `None` would suffice.
2. Consider using `and_then`:
`env::var(var_name).and_then(|v| v.parse::<usize>()).ok()`
3. and then to implement your rich logging, you can use [tap](https://docs.rs/tap/latest/tap/index.html), specifically `tap_err`.

Reply:
Sorry I turned on auto merge. Will address the feedback in another PR.

Reply:
No worries, those are stylistic recommendations, and aside from the first, not really binding.